compiler/src/dotty/tools/backend/jvm/GenBCode.scala

```diff
@@ -90,6 +94,20 @@ class GenBCode extends Phase { self =>
     try
       val result = super.runOn(units)
       generatedClassHandler.complete()
+      for holder <- ctx.asyncTastyPromise do
+        try
+          val asyncState = Await.result(holder.promise.future, 5.seconds)
```

Comment:
How about future.sequence?

Reply:
not sure I follow, there is only one future here - `ctx.asyncTastyPromise` returns an `Option[AsyncTastyHolder]` which holds a promise

Reply:
I see, I missed that.